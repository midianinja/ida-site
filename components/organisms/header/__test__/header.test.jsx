import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from '../header';

describe('Header', () => {
  afterEach(cleanup);
  
  it('renders without crashing', async () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId('ida-logo')).toHaveAttribute('src');
  });

  it('matches snapshot header', () => { 
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

