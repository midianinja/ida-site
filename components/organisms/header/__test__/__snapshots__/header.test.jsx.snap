// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot header 1`] = `
<header
  className="header"
>
  <img
    alt="Logo da Identidade Digital Ativista"
    className="logo"
    data-testid="ida-logo"
    src="/static/images/colorful-ida-logo.svg"
  />
  <div
    className="actionWrapper"
  >
    <button
      className="button __white __small"
      data-testid="button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Entrar
    </button>
    <button
      className="button __white __small"
      data-testid="button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Cadastre-se
    </button>
  </div>
</header>
`;
